src/plugins_types.h

```diff
@@ -181,6 +181,9 @@ void lysc_prefixes_free(struct lysc_prefix *prefixes);
  * must function as expected.
  *
  * Note that the \p value string is not necessarily zero-terminated. The provided \p value_len is always correct.
+ * Note that dynamically allocated value can be added to dictonary by function zero copy. This means that all store function
+ * have to free dynamically allocated value in case they dont used it. Also, in case of any error store function have to call free
```

Comment:
I suppose this is up for discussion but I always try to stick to the rule that if a function fails, it should not have any side-effect, if possible. That is why inserting the value into dictionary should be the last thing they perform. But this should be mentioned in the description.

Reply:
Then in original ligyang2 is two problems. First, if adding value to the dictionary by zero-copy function fails then the function release value "**https://github.com/CESNET/libyang/blob/libyang2/src/hash_table.c#L249-L253**". The second problem is some store function somewhere in the middle call function and add value to the dictionary. After that, they check if the value is valid. 

I'm not against it. It depends How we decide. In case when not functions are not going to releasing variable if an error occurs. I would have to rewrite plugins store functions.

Reply:
I agree with @radek-isa and tend to agree with this change. Seems to me that only this way it is always clear what the caller should do with the value (it is unknown in which point the callback failed), so I would prefer to accept this change.

Reply:
My idea was, of course, that is the callback fails, the value is unchanged, if it succeeds and the dynamic flag was used, the value was used and should not be freed (so the caller always knows what to do). I have just mentioned what is the best approach in my experience (I do not find freeing the value on error as the logical thing to do in general) but I do not really have anything against this solution, so fine.